File: CineBookDAL/Models/Booking.cs

using System;
using System.Collections.Generic;

namespace CineBookDataAccessLayer.Models;

public partial class Booking
{
    public int BookingId { get; set; }

    public int? UserId { get; set; }

    public int? ShowtimeId { get; set; }

    public decimal? TotalAmount { get; set; }

    public string? BookingStatus { get; set; }

    public DateTime? BookingTime { get; set; }

    public virtual ICollection<BookedSeat> BookedSeats { get; set; } = new List<BookedSeat>();

    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();

    public virtual Showtime? Showtime { get; set; }

    public virtual User? User { get; set; }
}

File: CineBookApp/src/app/interfaces/Booking.ts
import { BookedSeat } from "./BookedSeat";
import { Payment } from "./Payment";
import { Showtime } from "./Showtime";

export interface Booking {
  BookingStatus: any;
  bookingId: number;
  userId?: number;  
  bookingTime: string;  
  //bookedSeats: BookedSeat[];
  payments: Payment[];
  showtime: Showtime;
  seats: {
    seatNumber: string;
    rowLabel: string;
    price: number;
  }[];
  bookingStatus?: string;
  totalAmount?: any; 
 
}
